lib/array.rb

```diff
@@ -0,0 +1,24 @@
+class Array
+
+  #iterative inject - all inject use cases
+  def crush_down memo = self.shift, *arg, &block
```

Comment:
Unused method argument - `block`. If it's necessary, use `_` or `_block` as an argument name to indicate that it won't be used.